chapters/buffer_device_address.adoc

```diff
@@ -0,0 +1,181 @@
+====
+
+== Extension
+
+The original proposal was done with the  `VK_EXT_buffer_device_address` extension. Shortly afterwards, `VK_KHR_buffer_device_address` was added, which had some minor feature differences. Starting in Vulkan 1.2 this has become core and it is very widely avaiable on GPUs on every platform. Starting in Vulkan 1.3 it became required, so if you are using Vulkan 1.3, you are guaranteed support.
+
+=== Name alias
+
+The naming of this `Buffer Device Address` feature will be different depending on where you look.
+
+For GLSL, it is `buffer reference` (see link:https://github.com/KhronosGroup/GLSL/blob/main/extensions/ext/GLSL_EXT_buffer_reference.txt[GL_EXT_buffer_reference]).
+
+For SPIR-V, it is `PhysicalStorageBuffer` (see link:https://htmlpreview.github.io/?https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/KHR/SPV_KHR_physical_storage_buffer.html[SPV_KHR_physical_storage_buffer]). The name comes from the fact there was a `StorageBuffer` and now it is "Physical". (`Physical` is the opposite of `Logical` in SPIR-V and translates to basically "pointers" vs "no pointers")
+
+If you are coming from DirectX, you will know the feature as `GPU virtual address`.
+
+== Vulkan Side
+
+From the Vulkan code, the only 2 things you need to do is
+
+1. Add `VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT_KHR` when creating your `VkBuffer`
+2. Add `VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT_KHR` when allocating your `VkDeviceMemory`
+
+From here you can use the `vkGetBufferDeviceAddress` call and it will return a `VkDeviceAddress`. This is now your 64-bit pointer to that `VkBuffer` that can be handed down into your shader.
+
+[NOTE]
+====
+For tools, the `vkGetBufferDeviceAddress` function might break assumptions that Vulkan function only return `void` or `VkResult`
+====
+
+== SPIR-V Side
+
+[NOTE]
+====
+If you are using GLSL/HLSL/Slang/etc then the following is all taken care for you!
+====
+
+=== Capability
+
+The SPIR-V will contain the `OpCapability PhysicalStorageBufferAddresses` instruction that will match up with `VkPhysicalDeviceVulkan12Features::bufferDeviceAddress` (or `VkPhysicalDeviceBufferDeviceAddressFeatures::bufferDeviceAddress`) to let everyone know the device does support this.
+
+=== Addressing Model
+
+SPIR-V has an link:https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#Addressing_Model[Addressing Model] that has 3 group of options
+
+1. `Logical` - this is what Vulkan 1.0 uses, it has no concept of pointers
+2. `Physical32`/`Physical64` - this for OpenCL
+3. `PhysicalStorageBuffer64` - this is what you will use if you are making use of Buffer Device Address
+
+=== shaderInt64
+
+Since these `VkDeviceAddress` are represented as 64-bit interger pointers in your shader, you will likely want `shaderInt64` support for your device.
```

Comment:
```suggestion
Since these `VkDeviceAddress` are represented as 64-bit integer pointers in your shader, you will likely want `shaderInt64` support for your device.
```

Reply:
I don't think Int64 is strictly required.

Reply:
it is not required, I guess wanted to point that out that you can do it all without it if needed